### DISCUSSION TITLE
Consult about document reading and writing process

### DISCUSSION THREAD

**Discussion by dannyfe** [juicedata/juicefs#1200] [Q&A] [answered]
I want to know some detailed steps and definitions of the file reading process. For example, after understanding the writing process, what is the standard of using 32 < < 10 as a 32kb file measurement during reading, how to understand some calculation reasons of the offset during reading, and what is the ultimate purpose

**Comment by davies** [comment, marked as answer]
Where does the 32k come from? 

This page may be helpful: https://juicefs.com/docs/zh/community/internals/io_processing (no English version yet)

**Reply by dannyfe** [reply]
Before the read preview, the definition of "var lastbs Uint64 = 32 < < 10", thank you for your reply. I will refer to this link

**Reply by davies** [reply]
Some of storage format has a footer at the end of file, it's better to read more then request in one shot, so later read can reuse the result.

**Reply by dannyfe** [reply]
This seems to be a good deal of detail, I will continue to think about it
